federation/scim/src/main/java/org/keycloak/federation/scim/event/ScimBackgroundGroupMembershipUpdater.java

```diff
@@ -0,0 +1,72 @@
+package org.keycloak.federation.scim.event;
+
+import org.jboss.logging.Logger;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.timer.TimerProvider;
+import org.keycloak.federation.scim.core.ScimDispatcher;
+
+import java.time.Duration;
+
+/**
+ * In charge of making background checks and sent UPDATE requests from group for which membership information has changed.
+ * <p>
+ * This is required to avoid immediate group membership updates which could cause to incorrect group members list in case of
+ * concurrent group membership changes.
+ */
+public class ScimBackgroundGroupMembershipUpdater {
+    public static final String GROUP_DIRTY_SINCE_ATTRIBUTE_NAME = "scim-dirty-since";
```

Comment:
The group attribute will be visible as a regular group attribute and allow updating and removing the attribute when managing a group via Group API. Do we want to prevent that and enforce this attribute being managed only by the provider?

Wouldn't be better to add a property to the `GroupModel` (and a column in the database) to track the last time a group was updated? The idea is to have a better and indexable query to fetch dirty groups from the database by comparing the last time a group was updated with the last time the job ran. The latter can be stored in the component model of the provider, I think.

Reply:
I'm also fine if you think we can drop this scheduled task for now and focus on the core capabilities you are delivering. We can always revisit this later and start discussions to decide on the best solution for this problem before making the feature supported.

It should not be a blocker, IMO, and we should probably discuss it in a follow-up.